### DISCUSSION TITLE
[xstate/graph] v5 compatibility release and event filtering

### DISCUSSION THREAD

**Discussion by eahenke** [statelyai/xstate#4653] [Q&A]
I have two questions related to `@xstate/graph` as it relates to XState v5.

1. The latest version of `@xtate/graph`, `2.0.0-beta.5` refers in several places to `logic.getInitialState()` which is no longer a method of StateMachine/ActorLogic in XState v5, and causes a runtime error. It looks like this has actually been fixed in #4541 , but these changes have not been published in `@xstate/graph` yet. Is there more work waiting before the next beta release, and if so, is there an expected timeline for this release? Thank you!

2. Since snapshots no longer have an `event` property, is it still possible to filter by event in `getSimplePaths`? My use case is that I'm building a router, and want to validate that a state/route is accessible from the initial state based on context. I have a certain event `FORCE_ROUTE` that I'm adding on all states as a sort of homegrown [Routable States](https://github.com/statelyai/xstate/pull/4184) to handle some syncing w/ the window location. When validating the paths, I want to specifically ignore this event. 

In v4, it looked like:
```ts
const simplePaths = getSimplePaths(machine, {
    filter: (state) => {
      return state.event?.type !== EVENTS.FORCE_ROUTE;
    },
    stateSerializer: (state) => getCurrentPath(state),
  });
```

Is this behavior still possible in v5, or is there perhaps some other way I should be using to validate paths while still having some escape hatch events?  Thanks in advance!

**Comment by davidkpiano** [comment]
We still have to do some work updating `@xstate/graph` with XState v5, especially to make it work with other kinds of actors. Sorry for the delay on this!

**Reply by eahenke** [reply]
Thanks for the quick response - I really appreciate how responsive y'all are on these questions and issues. Just curious, do you have any rough guess on a timeline for the next graph release - weeks, months? I have proof-of-concept level code and I'm hoping to solidify it around v5 for a greenfield project, instead of doing that work for v4 and then upgrading. The route validation stuff isn't immediately necessary, so I could back burner that for a while if the graph updates are coming in the near-ish future.

**Reply by David-Botos** [reply]
Ah, @eahenke it looks like the docs now say its packaged by default with xstate: https://stately.ai/docs/graph

I hope this helps

**Reply by davidkpiano** [reply]
Yes that's right, thank you @David-Botos and apologies for not getting back to this earlier
